app/experimenter/nimbus-ui/src/components/PageEditAudience/index.tsx

```diff
@@ -48,9 +49,17 @@ const PageEditAudience: React.FunctionComponent<RouteComponentProps> = () => {
       }: Record<string, any>,
       next: boolean,
     ) => {
+      // issue #3954: Need to parse string IDs into numbers
+      const nimbusExperimentId = experiment.id;
+      let publish_status = experiment.publishStatus;
+      if (
+        experiment.isRollout &&
+        getStatus(experiment).live &&
+        experiment.populationPercent !== populationPercent
+      ) {
+        publish_status = NimbusExperimentPublishStatusEnum.DIRTY;
+      }
```

Comment:
Oh yeah see.  It would be so easy to mess this up on the client or change it and then have cases where fields are being changed but DIRTY isn't being set.  And it shouldn't even be dependent on any particular field or anything.  It should just be

1) If it's in LIVE/IDLE
2) Any mutation comes in
3) It's not LIVE/DIRTY